pinc/ProofreadingToolbox.inc

```diff
@@ -1,375 +1,111 @@
-            '[Sidenote:',
-            "top.surroundSelection('[Sidenote: ',']')"
-        );
-        $this->define_tool_link(
-            'illustration',
-            '[Illustration: ]',
-            '[Illustration:',
-            "top.surroundSelection('[Illustration: ',']')"
-        );
-        $this->define_tool_link(
-            'note',
-            '[** ]',
-            _('Note'),
-            "top.surroundSelection('[** ',']')"
-        );
-        $this->define_tool_link(
-            'brackets',
-            '[ ]',
-            '[ ]',
-            "top.surroundSelection('[',']')"
-        );
-        $this->define_tool_link(
-            'braces',
-            '{ }',
-            '{ }',
-            "top.surroundSelection('{','}')"
-        );
-        $this->define_tool_link(
-            'footnote',
-            '[Footnote #: ]',
-            '[Footnote #:',
-            "top.surroundSelection('[Footnote #: ',']')"
-        );
-        $this->define_tool_link(
-            'slash_star',
-            '/* */',
-            _('Start poetry/table'),
-            "top.surroundSelection('/*\\n','\\n*/')"
-        );
-        $this->define_tool_link(
-            'slash_number',
-            '/# #/',
-            _('Start block quote'),
-            "top.surroundSelection('/#\\n','\\n#/')"
-        );
-        $this->define_tool_link(
-            'thought_break',
-            '<tb>',
-            _('Thought Break'),
-            "top.surroundSelection('\\n<tb>\\n','')"
-        );
-        $this->define_tool_link(
-            'blank_page',
-            '[Blank Page]',
-            '[Blank Page]',
-            "top.replaceAllText('[Blank Page]')"
-        );
-    }
+        global $code_url;
+        $formatting = is_formatting_round($round);
 
-    private function array_extract( $source, $keys )
-    {
-        if ( $keys == 'ALL' )
+        echo "<div id='tools'>";
+        foreach ($this->tool_buttons as $tool_button)
         {
-            return $source;
+            $tool_button->draw($formatting);
         }
 
-        $result = array();
-        foreach ( $keys as $key )
+        draw_popup_link(_('Search/Replace'), 'srchrep.php', 'dp_searchrepl', 300, 250);
+        if ($formatting)
         {
-            if ( array_key_exists($key, $source) )
-            {
-                $result[$key] = $source[$key];
-            }
-            else
-            {
-                // It's tempting to do an assert(0), but $key could be for a tool
-                // that *can* exist, but happens not to in this installation.
-                // (i.e., currently, hieroglyph_transliterator)
-            }
+            spacer_bar();
+            draw_popup_link(_('Table Maker'), 'mktable.php', 'dp_mktable', 600, 500);
         }
-        return $result;
+        spacer_bar();
+        draw_popup_link(_('Greek Transliterator'), 'greek2ascii.php', 'gkasciiWin', 640, 250);
+
+        $url = get_faq_url($round->document);
```

Comment:
Ditto putting the doc links on a line by themselves

Reply:
And lets prefix them with "Documentation:" and remove the `[]`s (why were the brackets even there in the first place!?)

Reply:
Done this.